src/source/Ice/TurnConnection.c

```diff
@@ -629,32 +615,28 @@ STATUS turnConnectionAddPeer(PTurnConnection pTurnConnection, PKvsIpAddress pPee
 
     /* check for duplicate */
     CHK(turnConnectionGetPeerWithIp(pTurnConnection, pPeerAddress) == NULL, retStatus);
-    CHK_STATUS(doubleListGetNodeCount(pTurnConnection->turnPeerList, &peerCount));
-    CHK_WARN(peerCount < DEFAULT_TURN_MAX_PEER_COUNT, STATUS_INVALID_OPERATION, "Add peer failed. Max peer count reached");
+    CHK_WARN(pTurnConnection->turnPeerCount < DEFAULT_TURN_MAX_PEER_COUNT, STATUS_INVALID_OPERATION, "Add peer failed. Max peer count reached");
```

Comment:
Perhaps I am not understanding this. So you have an array of turn peers. It should get initialized to an initial value at first with status being set to something like not_existent. Then, you add a new peer. If we never "delete" a peer then adding it at the turnPeerCount is good but we might delete some turn peers making "holes" in the array right?

Reply:
we dont delete turn peers other than here when it fails to create transaction store due to OOM.